config/config.go

```diff
@@ -56,6 +58,18 @@ func IsCloudConfig(userdata string) bool {
 	return (header == "#cloud-config")
 }
 
+func IsMultipartMime(userdata string) bool {
+	bufioReader := bufio.NewReader(strings.NewReader(userdata))
+	textProtoReader := textproto.NewReader(bufioReader)
+	header, err := textProtoReader.ReadMIMEHeader()
+	if err != nil {
+		return false
+	}
+
+	contentType := header.Get("Content-Type")
+	return strings.Contains(contentType, "multipart/mixed")
```

Comment:
Shouldn't we run `ReadMIMEHeader` in a loop until we get an error or we find the `Content-Type` header?

Reply:
The ```multipart/mixed``` is a container for all the other mime types. It should be the first one. It also sets the ```boundary``` by which we parse the message. See:

https://github.com/flatcar/coreos-cloudinit/pull/21/files#diff-f1823e26b4f87b76a643597737f5496e0a01a47b14b411f56a350220c170d35e